ROSE2_geneMapper.py

```diff
@@ -405,35 +401,38 @@ def mapEnhancerToGeneTop(rankByBamFile, controlBamFile, genome, annotFile, enhan
                 distalGenes.remove(refID)
 
         # Now find the closest gene
+        closestGene = ''
         if len(allEnhancerGenes) == 0:
             closestGene = ''
         else:
             # get enhancerCenter
             enhancerCenter = (int(line[2]) + int(line[3])) / 2
 
-            # get absolute distance to enhancer center
-            distList = [abs(enhancerCenter - startDict[geneID]['start'][0])
-                        for geneID in allEnhancerGenes]
-            # get the ID and convert to name
-            closestGene = startDict[
-                allEnhancerGenes[distList.index(min(distList))]]['name']
+            try:
+                # get absolute distance to enhancer center
+                distList = [abs(enhancerCenter - list(startDict[geneID]['start'])[0]) for geneID in allEnhancerGenes]
+                
+                # get the ID and convert to name
+                closestGene = startDict[allEnhancerGenes[distList.index(min(distList))]]['name']
+            except:
```

Comment:
Do you want this to fail silently? Or give some meaningful message to the user? Or catch a specific kind of error that you expect?